src/qwed_new/api/main.py

```
@@ -1123,7 +1241,8 @@ async def register_agent(
             "message": "Agent registered successfully. Store the agent_token securely."
         }
     except Exception as e:
-        raise HTTPException(status_code=500, detail=str(e))
+        logger.error(f"Agent registration error: {redact_pii(str(e))}", exc_info=False)
+        raise HTTPException(status_code=500, detail="Agent registration failed")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)